src/snapshot_middleware/dir.rs

```diff
@@ -26,11 +26,19 @@ impl SnapshotMiddleware for SnapshotDir {
             return Ok(None);
         }
 
-        let children: Vec<VfsEntry> = entry.children(vfs)?;
+        let passes_filter_rules = |child: &VfsEntry| {
+            for rule in context.path_ignore_rules.iter() {
+                if !rule.passes(child.path()) {
+                    return false;
+                }
+            }
+
+            true
```

Comment:
Potentially more readable as:

```rs
context.path_ignore_rules.iter().all(|rule| rule.passes(child.path()))
```

Reply:
Ahah! I originally wrote this with iterators but chose `any` instead of `all` which did not work. I think you're right that this is cleaner.